### DISCUSSION TITLE
Error on upserting large number of txt files.

### DISCUSSION THREAD

**Discussion by Toolfolks** [FlowiseAI/Flowise#1826] [Q&A]
I have a Text File Input node.
I have loaded 37 txt files total size 16.3 meg.
When upserting I get this:
Error: TypeError: this.embeddings.embededDocuments is not a function

Is this related to the total file size / number of files ?

I have flowise installed on render with parameter 
FLOWISE_FILES_SIZE_LIMIT = 150mb

I have 1192 txt file in total that I eventually have to upload. How do I handle this ? Add multiple txt node. Does merging the txt files help or cause any issues with retrieving data?

Last question pinecone node now has Pinecode retriever & pinecone Vector Store. What are these parameters for please.
![WhatsApp Image 2024-02-27 at 15 58 32](https://github.com/FlowiseAI/Flowise/assets/125031610/12868bc1-4af2-4715-b681-15c6b89cd71f)

**Comment by anonix98** [comment]
I have the same problem.

**Reply by HenryHengZJ** [reply]
You need to have an embedding node connected to Pinecone.

**Reply by Toolfolks** [reply]
Please expand.

**Reply by edumoulin-oximae** [reply]
Your Pinecone node misses an embedding node ( the "Embedding" connector with an asterix under the "Document" connector) : you need to add an embedding node such as "OpenAI Embeddings" as you seem to use other OpenAI Nodes
![Capture d’écran 2024-04-05 à 15 54 22](https://github.com/FlowiseAI/Flowise/assets/104355762/41931864-8f28-487f-98a4-4f160aced6c5)
